edk2toollib/uefi/edk2/path_utilities.py

```diff
@@ -36,54 +46,52 @@ class Edk2Path(object):
 
     """
 
-    def __init__(self, ws: os.PathLike, package_path_list: Iterable[os.PathLike],
+    def __init__(self, ws: str, package_path_list: List[str],
```

Comment:
Does this need to be a list instead of a generic iterable? This particular argument is only being used with the iterator protocol in the for in loop.

Reply:
How we consume it can be any iterable, but due to the comment below, we always return it as a list - I thought it would be confusing for a user to pass in a package path list as a tuple, then when they retrieve it, they get a list. to avoid the confusion, I changed it to say we only allow a lists. I can change it back and just make it a bit more clear in the description of Edk2Path that the package path list is manipulated.

```python
        # Other code is dependent the following types, so keep it that way:
        #   - self.PackagePathList: List[str]
        #   - self.WorkspacePath: str
```

Reply:
> How we consume it can be any iterable, but due to the comment below, we always return it as a list - I thought it would be confusing for a user to pass in a package path list as a tuple, then when they retrieve it, they get a list. to avoid the confusion, I changed it to say we only allow a lists. I can change it back and just make it a bit more clear in the description of Edk2Path that the package path list is manipulated.
> 
> ```python
>         # Other code is dependent the following types, so keep it that way:
>         #   - self.PackagePathList: List[str]
>         #   - self.WorkspacePath: str
> ```

I think it's fine to allow a generic iterable and return a list as long as the documentation (type hints included) indicate so.

Reply:
Sounds good I've updated package_path_list type hint to be Iterable[str]. Thanks!